compiler/rustc_span/src/hygiene.rs

```diff
@@ -572,35 +578,32 @@ impl HygieneData {
         }
 
         // Reserve a new syntax context.
-        // The inserted dummy data can only be potentially accessed by nested `alloc_ctxt` calls,
-        // the assert below ensures that it doesn't happen.
```

Comment:
This comment is still relevant.
The dummy data is still inserted, and we still need to be careful for it to not be accessed from the nested calls.